### PR TITLE
Hosting.Testing readme

### PR DESCRIPTION
**Pull request by Tratcher**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4556)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHost.cs [hunk 1]
```diff
@@ -2,16 +2,19 @@
...
 using Microsoft.Extensions.Hosting;
+using Microsoft.Shared.DiagnosticIds;
 using Microsoft.Shared.Diagnostics;
 
 namespace Microsoft.Extensions.Hosting.Testing;
 
 /// <summary>
 /// Unit testing friendly configured host.
 /// </summary>
+[Experimental(diagnosticId: Experiments.Hosting, UrlFormat = Experiments.UrlFormat)]
```
Comment (joperezr) [inline src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHost.cs:17]: @geeknoid can you confirm that this is correct?

File: src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHostOptions.cs [hunk 2]
```diff
@@ -2,13 +2,16 @@
...
+using System.Diagnostics.CodeAnalysis;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Shared.DiagnosticIds;
 
 namespace Microsoft.Extensions.Hosting.Testing;
 
 /// <summary>
 /// Options to configure <see cref="FakeHost"/>.
 /// </summary>
+[Experimental(diagnosticId: Experiments.Hosting, UrlFormat = Experiments.UrlFormat)]
```
Comment (joperezr) [inline src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHostOptions.cs:14]: @geeknoid same here.

**Review by joperezr — COMMENTED** [review]
Other than making sure the Experimental attributes are right, this LGTM.

File: src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHost.cs (hunk 1 above)
Comment (Tratcher) [reply src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHost.cs:17]: Note these were added to match the manifest. If it's wrong, we'll need to update the manifest.
https://github.com/dotnet/extensions/blob/6fda2c8b8f8bacf74eaafff243a10ea20f849aaa/src/Libraries/Microsoft.Extensions.Hosting.Testing/Microsoft.Extensions.Hosting.Testing.json#L5-L6

Comment (joperezr) [reply src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHost.cs:17]: Got it, I just want to make sure we understand which one is correct, manifest or code. Given this is a Fake, it wasn't immediately obvious to me why this wouldn't be stable.

Comment (geeknoid) [reply src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHost.cs:17]: Oh actually, this is all working at expected.

The whole package is experimental as we have Stage=dev in the .csproj. Thus, the experimental attribute is being applied at the level of the assembly and doesn't need to be applied to each type/member. So you can safely remove this experimental annotation.

**Review by geeknoid — APPROVED** [review]
Just remove the changes to the .cs files please, those aren't needed.

**Review by geeknoid — APPROVED** [review]
Just remove the changes to the .cs files please, those aren't needed.

File: src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHost.cs (hunk 1 above)
Comment (joperezr) [reply src/Libraries/Microsoft.Extensions.Hosting.Testing/FakeHost.cs:17]: I see, that makes sense. I didn't know that we had infrastructure setup for injecting the attribute to the assembly but it does seem to be working then.

File: src/Libraries/Microsoft.Extensions.Hosting.Testing/README.md [hunk 3]
```diff
@@ -1 +1,52 @@
-README
+# Microsoft.Extensions.Hosting.Testing
+
+Tools for integration test hosting and related test oriented helpers.
```
Comment (danmoseley) [inline src/Libraries/Microsoft.Extensions.Hosting.Testing/README.md:3]: .. of ASP.NET Core apps?

also is it worth mentioning the Microsoft.AspNetCore.Testing package here, and vice versa in that readme?

Comment (danmoseley) [reply src/Libraries/Microsoft.Extensions.Hosting.Testing/README.md:3]: actually not necessarily of ASP.NET Core apps,  but of apps built with Microsoft.Extensions.Hosting. Anyway -- it should say it's not totally generic.

**Pull request merged**